orchestrator/src/main/kotlin/WorkerScheduleInfo.kt

```diff
@@ -0,0 +1,222 @@
+                context.workerRepositories.scannerJobRepository.create(context.ortRun.id, config)
+            }
+
+        override fun publishJob(context: WorkerScheduleContext, job: WorkerJob) {
+            context.publish(ScannerEndpoint, ScannerRequest(job.id))
+        }
+
+        override fun isConfigured(configs: JobConfigurations): Boolean =
+            configs.scanner != null
+    }
+
+private val evaluatorWorkerScheduleInfo =
+    object : WorkerScheduleInfo(EvaluatorEndpoint, runsAfter = listOf(AdvisorEndpoint, ScannerEndpoint)) {
+        override fun createJob(context: WorkerScheduleContext): WorkerJob? =
+            context.jobConfigs().evaluator?.let { config ->
+                context.workerRepositories.evaluatorJobRepository.create(context.ortRun.id, config)
+            }
+
+        override fun publishJob(context: WorkerScheduleContext, job: WorkerJob) {
+            context.publish(EvaluatorEndpoint, EvaluatorRequest(job.id))
+        }
+
+        override fun isConfigured(configs: JobConfigurations): Boolean =
+            configs.evaluator != null
+    }
+
+private val reporterWorkerScheduleInfo =
+    object : WorkerScheduleInfo(ReporterEndpoint, runsAfter = listOf(EvaluatorEndpoint)) {
+        override fun createJob(context: WorkerScheduleContext): WorkerJob? =
+            context.jobConfigs().reporter?.let { config ->
+                context.workerRepositories.reporterJobRepository.create(context.ortRun.id, config)
+            }
+
+        override fun publishJob(context: WorkerScheduleContext, job: WorkerJob) {
+            context.publish(ReporterEndpoint, ReporterRequest(job.id))
+        }
+
+        override fun isConfigured(configs: JobConfigurations): Boolean =
+            configs.reporter != null
+    }
+
+private val notifierWorkerScheduleInfo =
+    object : WorkerScheduleInfo(NotifierEndpoint, dependsOn = listOf(ReporterEndpoint)) {
+        override fun createJob(context: WorkerScheduleContext): WorkerJob? =
+            context.jobConfigs().notifier?.let { config ->
+                context.workerRepositories.notifierJobRepository.create(context.ortRun.id, config)
+            }
+
+        override fun publishJob(context: WorkerScheduleContext, job: WorkerJob) {
+            context.publish(NotifierEndpoint, NotifierRequest(job.id))
+        }
+
+        override fun isConfigured(configs: JobConfigurations): Boolean =
+            configs.notifier != null
+    }
+
+/**
+ * A map allowing access to the [WorkerScheduleInfo] objects for the different worker endpoints.
+ */
+internal val scheduleInfos = mapOf(
+    AnalyzerEndpoint.configPrefix to analyzerWorkerScheduleInfo,
+    AdvisorEndpoint.configPrefix to advisorWorkerScheduleInfo,
+    ScannerEndpoint.configPrefix to scannerWorkerScheduleInfo,
+    EvaluatorEndpoint.configPrefix to evaluatorWorkerScheduleInfo,
+    ReporterEndpoint.configPrefix to reporterWorkerScheduleInfo,
+    NotifierEndpoint.configPrefix to notifierWorkerScheduleInfo
+)
```

Comment:
I find it difficult to understand that some of the functions in `WorkerScheduleInfo` depend on this top-level property which in turn holds instances of `WorkerScheduleInfo`. It seems to be a candidate for using a sealed class or an enum class, especially as the list of valid worker schedule infos is pretty stable.

For example, using an enum class would by design show that there is limited list of pre-defined schedule info, and accessing them by prefix could be done like this:
```kotlin
enum class WorkerScheduleInfo(val configPrefix: String) {
    ANALYZER("analyzer"),
    SCANNER("scanner");

    companion object  {
        private val entriesByPrefix = entries.associateBy { it.configPrefix }

        operator fun get(configPrefix: String) = entriesByPrefix[configPrefix]
    }
}

fun test() {
    WorkerScheduleInfo["analyzer"]
}
```

Reply:
Good suggestion! I have added another commit which converts `WorkerScheduleInfo` to an enum class.